### DISCUSSION TITLE
KMemo: [ImageBlock] rotate the image 90 degrees

### DISCUSSION THREAD

**Discussion by radekjindra** [kryslt/KControls#59] [Q&A] [answered]
Hello, is it possible to rotate an image by 90 degrees in KMemo?

var
IB                    :  TKMemoImageBlock;
Protokol_ZKP :  TKmemo;

.......

if FileExists('./Data/'+LabeledEdit2.Text+' - '+LabeledEdit1.Text+'/Picture/3.jpg') then
 begin
   IB:=Protokol_ZKP.Blocks.AddImageBlock('./Data/'+LabeledEdit2.Text+' - '+LabeledEdit1.Text+'/Picture/3.jpg');
 end
else
 begin
  IB:=Protokol_ZKP.Blocks.AddImageBlock('./Picture/NoImage.jpg');
end;

.....
{something like}
IB.rotation :=90;

Thank you

**Comment by kryslt** [comment, marked as answer]
No, there is no such feature built into KControls.
For this you need to use an image manipulation library such as https://github.com/galfar/imaginglib or https://github.com/bgrabitmap/bgrabitmap.
If you need to rotate the image before inserting it into KMemo, use the image library before inserting the image into KMemo with AddImageBlock.
If you need to rotate an image after inserting it into KMemo, use the image library for the selected image block.
See kmemodlgimage.pas (and kmemofrm.pas) for how images can be manipulated with KMemo.
Take kmemodlgimage.pas as a reference for the more advanced image editing/manipulation form in KMemo.

**Reply by radekjindra** [reply]
Thanks a lot for the advice

**Comment by bird00101011** [comment]
Yes,I think he is right, before insert, rotate first. This is best method. And rotate image is easy.
________________________________
发件人: Radek Jindra ***@***.***>
发送时间: 2024年7月22日 16:32
收件人: kryslt/KControls ***@***.***>
抄送: Subscribed ***@***.***>
主题: Re: [kryslt/KControls] [ImageBlock] rotate the image 90 degrees (Discussion #59)


Thanks a lot for the advice

―
